core/src/main/java/io/questdb/cutlass/text/CairoTextWriter.java

```diff
@@ -254,8 +260,12 @@ private TableWriter openWriterAndOverrideImportTypes(
                         this.types.setQuick(i, BadDateAdapter.INSTANCE);
                         break;
                     case ColumnType.TIMESTAMP:
-                        logTypeError(i);
-                        this.types.setQuick(i, BadTimestampAdapter.INSTANCE);
+                        if (detectedType == ColumnType.DATE) {
+                            this.types.setQuick(i, typeManager.nextTimestampAdapter(false, timestampFormatFactory.get(DateFormatUtils.UTC_PATTERN), engine.getConfiguration().getTextConfiguration().getDefaultTimestampLocale()));
```

Comment:
it would be good to have a static copy of UTC pattern. `TimetampFormatUtils` already has static formatters. You should also use TimestampFormatUtils.enLocale for UTC.

The reason for using fixed locale is that UTC pattern does not contain any text to be re-interpreted, for example 'Tue'. I guess you can also try to pass `null` for the locale, but better avoid that. 

Hope this makes sense?